### PR TITLE
Oy2 19014 2

### PR DESCRIPTION
**Pull request by gitsierra**
## Purpose

Set up a framework for testing the new 2022 measures

https://qmacbis.atlassian.net/browse/MDCT-1839 
Here is the ticket for testing functionality that should be included in another test file not run on every measure

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

**Review by TheDanMiller — CHANGES_REQUESTED** [review]

File: services/ui-src/src/measures/2022/AMMAD/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    expect(V.validateAtLeastOneDeviationFieldFilled).toHaveBeenCalled();
+    expect(V.validateOneCatRateHigherThanOtherCatPM).toHaveBeenCalled();
+    expect(V.validateOneCatRateHigherThanOtherCatOMS).toHaveBeenCalled();
+    expect(V.validateNumeratorLessThanDenominatorOMS).toHaveBeenCalled();
+    expect(V.validateRateZeroOMS).toHaveBeenCalled();
+    expect(V.validateRateNotZeroOMS).toHaveBeenCalled();
+  });
+
+  jest.setTimeout(15000);
+  it("should not pass a11y tests", async () => {
```
Comment (TheDanMiller) [inline services/ui-src/src/measures/2022/AMMAD/index.test.tsx:214]: Not sure we need this test -- in this test if I am following we are verifying ally works and thats not on us

thoughts?

File: services/ui-src/src/measures/2022/AMMAD/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    useApiMock(apiData);
+    renderWithHookForm(component);
+    const badhtml = screen.getByTestId("measure-wrapper-form");
+    badhtml.append(document.createElement("button"));
+    const results = await axe(badhtml);
+    expect(results).not.toHaveNoViolations();
+  });
+
+  it("should not allow non state users to edit forms by disabling buttons", async () => {
```
Comment (TheDanMiller) [inline services/ui-src/src/measures/2022/AMMAD/index.test.tsx:223]: is this controlled at the measure level or somewhere else. If this code is the same for all measures then we should test it once where that logic happens if that makes sense. We can test it on the AMMAD measure at that level

**Review by TheDanMiller — DISMISSED** [review]
@gitsierra please ticket the moving of the read only test to somewhere more appropriate. It does not need done for each measure.

Did we ticket the fileupload and 409 tests somewhere else

File: services/ui-src/src/measures/2022/AMMAD/index.test.tsx (hunk 1 above)
Comment (abelzer212) [reply services/ui-src/src/measures/2022/AMMAD/index.test.tsx:214]: I don't think it's on us to make sure Ally works, but from testing I like making sure that we can make a test fail, and I'm not sure how else we would do that with our tests.

Comment (abelzer212) [reply services/ui-src/src/measures/2022/AMMAD/index.test.tsx]: I just pulled that one out

File: services/ui-src/src/measures/2022/AMMAD/index.test.tsx (hunk 2 above)
Comment (TheDanMiller) [reply services/ui-src/src/measures/2022/AMMAD/index.test.tsx:213]: The only thing I will say to that is testing it here is great but it makes that other part show up uncovered in reporting. 
@abelzer212

**Pull request merged**